### PR TITLE
chore: Override dependencies: netty -> 4.1.49.Final, jersey-common -> 2.31

### PR DESCRIPTION
**Pull request by AlanConfluent**
### Description 
This is effectively a rollback to our previous version of netty since the update to 4.1.63.Final in common is breaking master.  Also pins jersey-common and fixes a small bug breaking master

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -138,6 +138,7 @@
         <io.confluent.ksql.version>7.0.0-0</io.confluent.ksql.version>
         <io.confluent.schema-registry.version>${confluent.version.range}</io.confluent.schema-registry.version>
         <mockito.version>3.8.0</mockito.version>
+        <netty-tcnative-version>2.0.29.Final</netty-tcnative-version>
```
Comment (xjin-Confluent) [inline pom.xml:141]: https://mvnrepository.com/artifact/io.netty/netty-tcnative-boringssl-static/2.0.39.Final?

File: pom.xml [hunk 2]
```diff
@@ -475,6 +476,90 @@
                 <version>${scala.version}</version>
             </dependency>
 
+            <!-- Explicitly override all netty artifacts to get the version we want. Doing netty-all
+             doesn't seem to correctly override its individual dependencies, so all are enumerated
+             here -->
+            <dependency>
+                <groupId>io.netty</groupId>
+                <artifactId>netty-all</artifactId>
```
Comment (xjin-Confluent) [inline pom.xml:484]: Is it safe then to remove netty-all?

**Review by xjin-Confluent — APPROVED** [review]
Please confirm the version of boring ssl. 
Othereise LGTM

File: pom.xml (hunk 1 above)
Comment (AlanConfluent) [reply pom.xml:141]: Oops, meant to update that.  Changed.

File: pom.xml (hunk 2 above)
Comment (AlanConfluent) [reply pom.xml:484]: Yeah, it is.  Removed.

**Comment by AlanConfluent** [comment]
I chatted with @xjin-Confluent about repurposing this PR to make it a roll back to an older version rather than pushing forward.

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java [hunk 3]
```diff
@@ -335,8 +335,8 @@ static boolean isConnectErrorResponse(final JsonObject ksqlEntity) {
           source.getString("valueFormat"),
           formatQueries(source.getJsonArray("readQueries")),
           formatQueries(source.getJsonArray("writeQueries")),
-          Optional.ofNullable(source.getString("timestamp")),
-          Optional.ofNullable(source.getString("windowType")),
+          Optional.ofNullable(emptyToNull(source.getString("timestamp"))),
+          Optional.ofNullable(emptyToNull(source.getString("windowType"))),
```
Comment (agavra) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java:339]: is this pertinent to this PR?

**Review by agavra — COMMENTED** [review]
LGTM if the tests pass, though might be good to document here why netty was bumped in first place. I just want to make sure it wasn't related to some CVE

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java (hunk 3 above)
Comment (AlanConfluent) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java:339]: No but someone introduced a small bug in the meanwhile.

Comment (mjsax) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java:339]: Seems to overlap with https://github.com/confluentinc/ksql/pull/7475

Comment (AlanConfluent) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java:339]: Ah, yeah.  I was just trying to fix everything in one fell swoop.

**Review by agavra — APPROVED** [review]
Huge thanks @AlanConfluent!

**Pull request merged**